core/src/test/scala/unit/kafka/admin/ConsumerGroupServiceTest.scala

```diff
@@ -106,7 +106,8 @@ class ConsumerGroupServiceTest {
   }
 
   private def listGroupOffsetsResult: ListConsumerGroupOffsetsResult = {
-    val offsets = topicPartitions.map(_ -> new OffsetAndMetadata(100)).toMap.asJava
+    // Half of the partitions of the testing topics are set to have a negative integer offset (null value [KAFKA-9507 for reference])
+    val offsets = topicPartitions.zipWithIndex.map{ case (tp, i) => tp -> ( if(i % 2 == 0) null else new OffsetAndMetadata(100) ) }.toMap.asJava
```

Comment:
I am not comfortable with hijacking all existing tests like this. Would it be possible to define a new unit test(s) which triggers the identified issue(s)? I think that it will help to not regress in the future if we have explicit test(s).

Reply:
Yes of course. Makes sense to keep tests isolated to avoid possible conflicts on the future. Will do that.